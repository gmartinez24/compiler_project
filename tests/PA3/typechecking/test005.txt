main

int x, y;

/* 
 * not all paths return - while this is undesirable, 
 * unfortunately this is not something we can detect now 
 */
function max (int a, int b) : int {
    if (a > b) then
        return a;
    else 
        // return b;
        a = b; 
    fi;
};

/* 
 * However, we can still detect if there a branch returning wrong type
 * Even though this is not comprehensive - we need a static analysis
 * to have better coverage
 */

function coolFunc(int a) : int{
    if ( a > x ) then
        return a;
    else
        return ( a > x ); // should trigger error here
    fi;
};


{
    x = 5;
    y = 3;
    call printInt(call max(x, y));
}.