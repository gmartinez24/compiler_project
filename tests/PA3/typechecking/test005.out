Error type-checking file.
TypeError(28,9)[Function coolFunc returns bool instead of int.]

